data/redisrepo/order_iter.go

```diff
@@ -0,0 +1,82 @@
+package redisrepo
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/orbs-network/order-book/models"
+	"github.com/orbs-network/order-book/utils/logger"
+	"github.com/orbs-network/order-book/utils/logger/logctx"
+)
+
+// ////////////////////////////////////////////////
+type OrderIter struct {
+	index int
+	ids   []string
+	redis *redisRepository
+}
+
+func (i *OrderIter) Next() *models.Order {
+	ctx := context.Background()
+
+	if i.index >= len(i.ids) {
+		logctx.Error(ctx, "Error iterator reached last element")
+		return nil
+	}
+
+	// increment index
+	i.index = i.index + 1
+	// get order
+	orderId, err := uuid.Parse(i.ids[i.index])
+	if err != nil {
+		logctx.Error(ctx, "Error parsing bid order id", logger.Error(err))
+		return nil
+	}
+	order, err := i.redis.FindOrderById(ctx, orderId)
+	if err != nil {
+		logctx.Error(ctx, "Error fetching order", logger.Error(err))
+		return nil
+	}
+
+	return order
+}
+
+func (i *OrderIter) HasNext() bool {
+	return i.index < (len(i.ids) - 1)
+}
+
+// ////////////////////////////////////////////////
+func (r *redisRepository) GetMinAsk(ctx context.Context, symbol models.Symbol) models.OrderIter {
+	key := CreateSellSidePricesKey(symbol)
+
+	// Min ask price for selling
+	orderIDs, err := r.client.ZRange(ctx, key, 0, -1).Result()
+	if err != nil {
+		logctx.Error(ctx, "Error fetching asks", logger.Error(err))
+	}
+	// create order iter
+	return &OrderIter{
+		index: -1,
```

Comment:
Why initialise this with -1? Better to start with 0 so it aligns with how we usually think about array indexes?

Reply:
i like the use of -1 as in "not been used" rather than 0 which is more implicit in case of only one order in the array